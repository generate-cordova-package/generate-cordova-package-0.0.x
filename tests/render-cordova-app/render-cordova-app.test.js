/* global it, expect */
/* (see <https://github.com/standard/standard/issues/905>) */

const renderCordovaAppTree = require('../../lib/cordova-app/render')

const renderCordovaAppProperties = require('./render-cordova-app-properties.config')

it('renders fs-tree object correctly for cordova-app', () => {
  const outputTree = renderCordovaAppTree(renderCordovaAppProperties)

  // using JSON.stringify in this test case to help verify that
  // lib/helpers/xml-component-to-string.js handles invisible characters
  // such as unwanted carriage return (\r) & correct newline (\n)
  // all correctly
  expect(JSON.stringify(outputTree)).toMatchSnapshot()
})
