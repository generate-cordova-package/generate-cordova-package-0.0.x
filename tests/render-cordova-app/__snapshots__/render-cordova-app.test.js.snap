// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders fs-tree object correctly for cordova-app 1`] = `"{\\"test-app\\":{\\"README.md\\":\\"# test-app\\\\n\\\\nCopyright 2023-present Alice <alice@alice.com>\\\\n\\",\\"config.xml\\":\\"<widget id=\\\\\\"cc.testapp\\\\\\" version=\\\\\\"0.1.1\\\\\\">\\\\n  <name>test-app</name>\\\\n  <description>test app</description>\\\\n</widget>\\\\n\\",\\"package.json\\":\\"{\\\\n  \\\\\\"name\\\\\\": \\\\\\"test-app\\\\\\",\\\\n  \\\\\\"version\\\\\\": \\\\\\"0.1.1\\\\\\",\\\\n  \\\\\\"// copyright\\\\\\": \\\\\\"Copyright 2023-present Alice <alice@alice.com>\\\\\\",\\\\n  \\\\\\"description\\\\\\": \\\\\\"test app\\\\\\",\\\\n  \\\\\\"license\\\\\\": \\\\\\"ISC\\\\\\",\\\\n  \\\\\\"author\\\\\\": {\\\\n    \\\\\\"name\\\\\\": \\\\\\"Alice\\\\\\",\\\\n    \\\\\\"email\\\\\\": \\\\\\"alice@alice.com\\\\\\"\\\\n  }\\\\n}\\\\n\\",\\"www\\":{\\"index.html\\":\\"<html>\\\\n  <header>\\\\n    <link rel=\\\\\\"stylesheet\\\\\\" href=\\\\\\"body.css\\\\\\"/>\\\\n  </header>\\\\n  <body>\\\\n    <h1>test-app</h1>\\\\n    <div id=\\\\\\"status\\\\\\"/>\\\\n    <script src=\\\\\\"cordova.js\\\\\\"></script>\\\\n    <script src=\\\\\\"app.js\\\\\\"></script>\\\\n  </body>\\\\n</html>\\\\n\\",\\"app.js\\":\\"// Copyright 2023-present Alice <alice@alice.com>\\\\n\\\\ndocument\\\\n  .getElementById('status')\\\\n  .appendChild(document.createElement('b'))\\\\n  .appendChild(document.createTextNode('status:'))\\\\ndocument.addEventListener('deviceready', function () {\\\\n  document\\\\n    .getElementById('status')\\\\n    .appendChild(document.createElement('i'))\\\\n    .appendChild(document.createTextNode('ready'))\\\\n})\\\\n\\",\\"body.css\\":\\"body {\\\\n  background-color: #fff;\\\\n}\\\\n\\"}}}"`;
